Fixing infinite spinning animation on user menu ctrl+click

Faldon opened the pull request:
Resembling the behaviour of the app menu, the user menu now closes on ctrl+click and does not show the loading spinner.

Fixes #616

rullzer committed:
Group shares with same source and target

Fixes #24575

Note that this is a very limited solution and eventually we want smarter
merging!

jvillafanez committed:
Close session fast when doing a test to prevent blocking the next
requests

Commit:
Add integration tests for merging received shares

Commit:
Improved share grouping readability + fixed test

Commit:
Add repair step for unmerged shares (WIP)

Commit:
Added more tests for sharing's MountProvider

Commit:
Group incoming shares for resharing in JS

Commit:
Adjust repair version check for unmerged shares

rullzer committed:
Convert Share API to use the AppFramework

rullzer committed:
Remove unneeded wrapper

rullzer committed:
Throw OCSNotFoundExceptions

rullzer committed:
Add OCSShareAPIMiddleware

* This will cleanup the locks after each request
* Move check for enabled share api to the middleware

rullzer committed:
More exceptions

rullzer committed:
DataResponse required ['data' => DATA]

jvillafanez committed:
Fix file permissions for SMB (read-only folders will be writeable) (#25301)

* Fix file permissions for SMB (read-only folders will be writeable)

* Read-only folders won't be deletable

* Added comment for the read-only behaviour for folders

rullzer committed:
Fix tests

Commit:
Fix RepairUnmergedShares to not skip valid repair cases

The repair step was a bit overeager to skip repairing so it missed the
case where a group share exists without subshares but with an
additional direct user share.

Commit:
Add integration tests for double shares with rename in between

rullzer committed:
Add OCSShareAPIMiddleware tests

Commit:
Make share target consistent when grouping group share with user share

In some situations, a group share is created before a user share, and
the recipient renamed the received share before the latter is created.
In this situation, the "file_target" was already modified and the second
created share must align to the already renamed share.

To achieve this, the MountProvider now groups only by "item_source"
value and sorts by share time. This makes it so that the least recent
share is selected as super-share and its "file_target" value is then
adjusted in all grouped shares.

This fixes the issue where this situation would have different
"file_target" values resulting in two shared folders appearing instead
of one.

nickvergessen committed:
Make sure each plugin is only added once

nickvergessen committed:
Add the systemtag js files

nickvergessen committed:
Add system tag check

nickvergessen committed:
Add file size as check

juliusknorr committed:
Change app menu to white

nickvergessen committed:
Add file mime type

jancborchardt committed:
up the minimum size of the public share video a bit, use 16/9 ratio, fix #603

nickvergessen committed:
User agent check

nickvergessen committed:
Add Request URL

nickvergessen committed:
Add remote address

nickvergessen committed:
Add request time

nickvergessen committed:
Do not allow to delete/update group shares as a group member

jancborchardt committed:
adjust opacity of icons in apps and user menu

juliusknorr committed:
Move to svg filter on app menu to support IE9+

stweil committed:
Fix misspelling of database "READ COMMITTED"

Signed-off-by: Stefan Weil <sw@weilnetz.de>

DeepDiver1975 committed:
Adding quota plugin to new dav endpoint (#25615)

* Adding quota plugin to new dav endpoint

* Added integrated test failing in old endpoint

* Added 0B quota test

MorrisJobke committed:
add a UI to render proper time picker

Commit:
Fixed user menu to not show spinner on ctrl+click.

Fixes #616

Commit:
Fixed user menu to not show spinner on ctrl+click.

Fixes #616

schiessle committed:
Merge pull request #648 from nextcloud/add-quotaplugin-v2-master

[upstream] Adding quota plugin to new dav endpoint

jancborchardt committed:
slight animation for the sidebar for better feedback what happens

MorrisJobke committed:
Merge pull request #651 from stweil/master

Fix misspelling of database "READ COMMITTED"

MorrisJobke commented on core/js/js.js:
```diff
@@ -1519,14 +1519,21 @@ function initCore() {
-			$page.find('img').remove();
-			$page.find('div').remove(); // prevent odd double-clicks
-			$page.prepend($('<div/>').addClass('icon-loading-small-dark'));
+			if(!event.ctrlKey) {
```
On OS X this is `event.metaKey` could you also check for this?

MorrisJobke replied:
And also adjust the other menu too :)

rullzer committed:
Dark hackery to not always disable CSRF for OCS controllers

Commit:
Added OS X meta key check for stopping spinning animation in app and user menu.

Commit:
Added OS X meta key check for stopping spinning animation in app and user menu.

icewind1991 commented:
How about middle mousebutton?

MorrisJobke commented:
> How about middle mousebutton?

On a trackpad there is no middle mousebutton 😝 



@Faldon Thanks! Works really nice now :)

MorrisJobke commented:
👍

jancborchardt committed:
Merge pull request #625 from nextcloud/video-container-size

up the minimum size of the public share video a bit, use 16/9 ratio

jancborchardt committed:
fix shading of apps icon in app menu

rullzer committed:
Merge pull request #656 from nextcloud/ocs_controller_csrf

Dark hackery to not always disable CSRF for OCS controllers

schiessle committed:
Merge pull request #627 from nextcloud/white-app-menu

Change app menu to white background with dark icons

schiessle commented:
> On a trackpad there is no middle mousebutton

Not everyone uses a trackpad and some people have a well configured track pack, e.g. I can do a middle click by tabbing with three fingers :wink: 

Also it is not a either or... I would love if we could extend the behavior for middle click as well.  @Faldon do you think you can add this? Thanks!

Commit:
Transifex Owncloud > Nextcloud

Change transifex link Owncloud > Nextcloud

juliusknorr committed:
Fix wrong preserveAspectRatio at app menu icons

Faldon commented:
Will look at it tomorrow. ;-)

juliusknorr committed:
Merge pull request #662 from thelinuxfr/patch-1

Transifex Owncloud > Nextcloud

nextcloud-bot committed:
[tx-robot] updated from transifex

MariusBluem committed:
Merge pull request #664 from nextcloud/white-app-menu-fix

Fix wrong preserveAspectRatio at app menu icons

MorrisJobke commented:
> Not everyone uses a trackpad and some people have a well configured trackpack, e.g. I can do a middle click by tapping with three fingers 😉

I read it as question directly to me 🙈 Sorry, just misunderstood this one.

jancborchardt commented:
cc @ChristophWurst here cause of the good timing with same pull request at https://github.com/nextcloud/server/pull/661 ;D We should continue here.

Faldon commented:
Hmmm... Using Chrome on Arch Linux, I cannot even detect the middle mouse click.
I get the event.which === 1 for left button click.
When I click the menu with middle mouse button, no spinner appears and menu does not close.

sndrr committed:
Update Upgrade.php

sndrr committed:
Rebranded to Nextcloud for the occ command line tool

juliusknorr committed:
Add animation to app-content

sndrr committed:
Revert (c) back to ownCloud, Inc.
To be dealt with later

juliusknorr committed:
Fix javascript tests for sidebar animation

jancborchardt committed:
remove extra file size padding to fix action menu position

Commit:
typo

nextcloud-bot committed:
[tx-robot] updated from transifex

rullzer committed:
Merge pull request #675 from nextcloud/enoch85-patch-1

typo

blizzz committed:
Merge pull request #595 from nextcloud/fix_slow_test

Close session fast when doing a test to prevent blocking the next requests

nextcloud-bot committed:
[tx-robot] updated from transifex

Faldon commented:
So, did some more tests:
It appears, that the click event is not raised on middle button and right button click. At least not in Chrome  on Arch and Fedora.
For testing, I set the event to prevent the default immediately after the event occurs. While this successfully stopped navigating on left button click (and decorating every menu entry with spinning icons :-D ), on middle button click my browser just opened the page in a new tab. No event handling after all.

rullzer committed:
Add OCS-APIREQUEST header to intergration test calls

jancborchardt committed:
fix missing semicolon to fix themed logo on log in page

rullzer committed:
We should properly check for 'true' instaed of the bool

nickvergessen committed:
Validate the input of the theming options

LukasReschke committed:
Merge pull request #674 from nextcloud/action-menu-position

remove extra file size padding to fix action menu position

nextcloud-bot committed:
[tx-robot] updated from transifex

MorrisJobke committed:
Merge pull request #672 from sndrr/occ-rebrand-to-nextcloud

Occ rebrand to nextcloud

schiessle committed:
Merge pull request #685 from nextcloud/validate-theming-input

Validate the input of the theming options

LukasReschke committed:
Merge pull request #684 from nextcloud/fix_csrf_ocs

Fix OCS CSRF

nickvergessen committed:
Move Share backends to PSR-4 instead of using class path (#24941)

nickvergessen committed:
Load the timezones via select2

nickvergessen committed:
Translate the errors

nickvergessen committed:
Tags need to be loaded before

MorrisJobke committed:
Merge pull request #653 from nextcloud/animate-sidebar

slight animation for the sidebar for better feedback what happens

MorrisJobke commented:
Let's get this in then :)

@juliushaertl @ChristophWurst Mind to review?

MorrisJobke committed:
Merge pull request #682 from nextcloud/fix-theming-logo

fix missing semicolon to fix themed logo on log in page

MorrisJobke committed:
Shake the login fields if password is wrong

juliusknorr committed:
Apply inverted app icon style to newly installed icons

MorrisJobke committed:
Merge pull request #695 from nextcloud/shake-wrong-password

Shake the login fields if password is wrong

nextcloud-bot committed:
[tx-robot] updated from transifex

nickvergessen committed:
Allow to define the operation

ChristophWurst committed:
add invalidateOldTokens to IProvider interface

nickvergessen committed:
No multi support, less magic

rullzer committed:
Add intergration test

nickvergessen committed:
Ooops

MorrisJobke committed:
Merge pull request #700 from nextcloud/token-provider-invalidateoldtokens

add invalidateOldTokens to IProvider interface

MorrisJobke committed:
Merge pull request #637 from nextcloud/h1-prevent-delete-update-on-group-shares

Do not allow to delete/update group shares as a group member

MorrisJobke committed:
Merge pull request #701 from nextcloud/fix-add-apps-icon

Apply black app icon style to newly installed app icons

nextcloud-bot committed:
[tx-robot] updated from transifex

MorrisJobke committed:
Add workflowengine to check-code call

MariusBluem committed:
Merge pull request #709 from nextcloud/add-code-check

Add workflowengine to check-code call

rullzer committed:
Merge pull request #474 from nextcloud/group_shares

Group shares with same source and target

michag86 committed:
Apply password policy on user creation

jancborchardt commented:
@Faldon any update here? :)

nickvergessen committed:
Merge pull request #611 from nextcloud/add-more-checks

Add more checks

nickvergessen committed:
Add access control and automated tagging as shipped

jurgenhaas committed:
Exit from upgrade without reporting error

As reported at https://help.nextcloud.com/t/silent-mode-for-occ-upgrade/2120 we should exit with a return value of 0 (zero) if the current instance is already up to date so that automatic deployment tools don't recognize that as an error.

nickvergessen committed:
Merge pull request #717 from nextcloud/mark-apps-as-shipped

Add access control and automated tagging as shipped

jancborchardt committed:
switch nested containers in update to semibold style

nextcloud-bot committed:
[tx-robot] updated from transifex

Faldon commented:
Sorry, was a little busy.
With the changes @ChristophWurst proposed, I can reproduce, that middle-clicking and cttl-/meta-clicking does not open the links anymore (though it closes the menu now.)

But I think I have got a proper solution:

```
@@ -1520,12 +1523,12 @@
 		var $menu = $('#header #settings');
 
 		// show loading feedback
-		$menu.delegate('a', 'click', function(event) {
+		$menu.delegate('a', 'mousedown', function(event) {
 			var $page = $(event.target);
 			if (!$page.is('a')) {
 				$page = $page.closest('a');
 			}
-			if(!event.ctrlKey && !event.metaKey) {
+			if(event.which === 1 && !event.ctrlKey && !event.metaKey) {
 				$page.find('img').remove();
 				$page.find('div').remove(); // prevent odd double-clicks
 				$page.prepend($('<div/>').addClass('icon-loading-small-dark'));
@@ -1533,7 +1536,12 @@
 				// Close navigation when opening menu entry in
 				// a new tab
 				OC.hideMenus();
+				// On middle click or on first button click with ctrl key or meta key hold
+				if(event.which === 2 || (event.which === 1 && (event.ctrlKey || event.metaKey))) {
+					window.open($page, '_blank');
-			}
+				}
+			}
+			$($page).click();
 		});
 	}
```

It's a little bit more logic, but handles opening links only if the primary button was clicked, with or without ctrl key or meta key modifier, or if the middle mouse button was clicked.
I will push it for testing (I have a mac keyboard but it is attached to Arch Linux. :-) )

Commit:
Changed app and user menu delegates to mousedown events.
Added proper handling of primary mouse button click with and without ctrl-/meta-key modifier.
Added handlig of middle mouse button click.

Commit:
Changed app and user menu delegates to mousedown events.
Added proper handling of primary mouse button click with and without ctrl-/meta-key modifier.
Added handlig of middle mouse button click.

nickvergessen committed:
Merge pull request #718 from jurgenhaas/patch-1

Exit from upgrade without reporting error

jancborchardt committed:
reduce info on update screens, introduce button to refresh

Commit:
fix #724 by using step instead of job

LukasReschke committed:
[master] Add new apps as official

LukasReschke committed:
Merge pull request #730 from nextcloud/add-new-apps-as-official

[master] Add new apps as official

nickvergessen committed:
Order alphabetically and deduplicate the shipped.json

nickvergessen committed:
Merge pull request #729 from nextcloud/fix-724

Fix #724 by using step instead of job

blizzz committed:
Correct name in Console Application.php

see changes :)

MorrisJobke committed:
Merge pull request #465 from nextcloud/fix_smb_attributes

Fix file permissions for SMB (read-only folders will be writeable) (#…

nickvergessen committed:
Increment the versions and adjust the capitilization

nextcloud-bot committed:
[tx-robot] updated from transifex

nickvergessen committed:
Merge pull request #737 from nextcloud/blizzz-fix-name-console

Correct name in Console Application.php

juliushaertl commented:
Code looks good and works fine for me, please check for the failing drone build. @Faldon

Faldon commented:
@juliushaertl I have to confess, I'm somewhat stuck with the drone output. I can determine, there is one failed feature test in sharing-v1 regarding permissions not registered in OC_Filesystem.

rullzer committed:
Merge pull request #732 from nextcloud/fix-shipped.json

Order alphabetically and deduplicate the shipped.json

rullzer committed:
Disbale failing intergration test for now

I have not been able to reproduce this failure locally yet.
But it seems to happen often on CI.
So disable for now.

rullzer committed:
Merge pull request #747 from nextcloud/up_24941

Move Share backends to PSR-4 instead of using class path (#24941)

MorrisJobke commented:
It's weird ... this has failed on others PRs too. :(

@rullzer Do you have any idea how this works?

rullzer commented:
See: https://github.com/nextcloud/server/pull/752

nickvergessen committed:
More casing

nickvergessen committed:
Disable by default

nextcloud-bot committed:
[tx-robot] updated from transifex

nextcloud-bot committed:
[tx-robot] updated from transifex

Commit:
Add black version of system tags icon

Includes PNG (16x16px) and SVG (scoured) versions in black, to be uniform with the other icon files.

schiessle committed:
Merge pull request #711 from nextcloud/password-policy-on-createUser

Apply password policy on user creation

LukasReschke committed:
Merge pull request #728 from nextcloud/update-styling

Design fixes for updates

LukasReschke committed:
Merge pull request #752 from nextcloud/disbale_failing_test

Disbale failing intergration test for now

juliushaertl commented:
#752 is merged, @Faldon could you rebase your pullrequest?

LukasReschke committed:
Merge pull request #445 from nextcloud/ocs_share_to_appframework

OCS Share API to appframework

rullzer committed:
Fix intergration test for admin_audit

LukasReschke committed:
Also cache WOFF, SVG and GIF

MorrisJobke committed:
Merge pull request #739 from nextcloud/increment-versions-master

Increment the versions and adjust the capitilization

LukasReschke committed:
Merge pull request #769 from nextcloud/also-cache-fonts-svg-and-gif

Also cache WOFF, SVG and GIF

MariusBluem committed:
Merge pull request #764 from nextcloud/fix-tags-icon

Add black version of system tags icon

Faldon committed:
Merge branch 'user_menu_ui' of https://github.com/Faldon/server into user_menu_ui

Faldon commented:
I hope this was right -.-

MorrisJobke commented:
> I hope this was right -.-

Doesn't seem so. Should I create a new PR with only your changes?

MorrisJobke commented:
I tried to resolve this here: #778

jancborchardt commented:
Closing in favor of that new PR then.
